### PR TITLE
fix(#10751): support case-insensitive and truthy values for unique_tel validation

### LINKED ISSUE #10751: Duplicate phone number validation broken when using "<phone cht:unique_tel='TRUE'/>"
**Describe the bug**
Duplicate phone number validation broken when using `<phone cht:unique_tel="TRUE"/>`

**To Reproduce**

1. Use [phone-number-input](https://docs.communityhealthtoolkit.org/building/forms/app/#phone-number-input) widget on a person-create contact form
2. Make sure you have the namespaces column in the “settings” tab of your form populated with a value that includes `cht=https://communityhealthtoolkit.org`
3.  In the “survey” tab, add `instance::cht:unique_tel` column and set the value to `TRUE` for the phone number field. (Spreadsheet editors such as MS Excel and Google Sheets convert `true` to `TRUE` automatically, unless you write force text format with `'true`)
4. Open the form as the user.
5. Phone validation does not work. It allows adding a duplicate number.

**Expected behavior**
Validation should work with `TRUE` or `True`, regardless of the case.
It works for other ODK columns such as `read_only`.

**Environment**
- Instance: N/A
- Browser: any
- Client platform: any
- App: webapp
- Version: 4.11.0+ (where `instance::cht:unique_tel` is supported)

**Additional context**
I think we need to make a case-insensitive check for `true` here:
https://github.com/medic/cht-core/blob/589a9d49a6bc9f8a8af2dc38447b9265c736a5b1/webapp/src/js/enketo/widgets/phone-widget.js#L61

---

## Good first issue

This ticket has the [Good first issue label](https://github.com/medic/cht-core/issues?q=state%3Aopen%20label%3A%22Good%20first%20issue%22)! This means it's been especially curated by other CHT contributors to be easy to work on for first time contributors [per the docs](/community/contributing/creating-good-first-issues/).

To succeed on this ticket, please:
1. Expand the `$wrapper.attr('data-cht-unique_tel') === 'true'` check in the phone-widget to include the other cases @jkuester mentioned below.
2. Add [unit tests](https://github.com/medic/cht-core/blob/589a9d49a6bc9f8a8af2dc38447b9265c736a5b1/webapp/tests/karma/js/enketo/widgets/phone-widget.spec.ts) for the new cases.

### PR DESCRIPTION
**Pull request by vikrantwiz02**
# Description
This PR resolves an issue where the cht:unique_tel validation was strictly case-sensitive. Following maintainer feedback, the truthy/falsy logic has been centralized into a core library to ensure consistency across the CHT and alignment with global ODK standards. This prevents validation bypasses caused by common spreadsheet auto-formatting (e.g., "True" instead of "true").

**Changes:**
- New Utility Library: Created webapp/src/js/enketo/lib/pyxform-boolean.js to handle standardized pyxform truthy/falsy values (true, yes, true()).
- Source Documentation: Included a reference link to the official pyxform constants in the library comments to "persist the connection" to the source of truth.
- Widget Refactor: Updated the phone widget to utilize this centralized logic, making unique_tel validation case-insensitive and more robust.
- Clean History: Flattened the branch into a single clean commit on top of the latest master.

**Validation Done:**
I ran the full Enketo regression suite within the webapp/ directory to ensure zero side effects across the platform.
cd webapp
npm run unit -- --watch=false --include='../tests/karma/js/enketo/**/*.spec.ts'

Result: 103/103 tests passed (This includes the new suite for the pyxform-boolean library and the updated phone widget tests)

# Issue Number
Fixes #10751

# AI Disclosure
I used GPT-5.3-Codex to assist in architecting the unit test edge cases and ensuring ODK-standard compliance.

# Code review checklist

- [ ] UI/UX backwards compatible: Test it works for the new design (enabled by default). And test it works in the old design, enable `can_view_old_navigation` permission to see the old design. Test it has appropriate design for RTL languages. 
- [x] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/)
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [x] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [x] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.
- [x] AI disclosure: Please disclose use of AI per [the guidelines](https://docs.communityhealthtoolkit.org/community/contributing/ai-guidelines/).

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/js/enketo/widgets/phone-widget.js [hunk 1]
```diff
@@ -49,6 +49,17 @@ const deprecated = {
   isDeprecated: ($wrapper) => !$wrapper.hasClass('or-appearance-tel'),
 };
 
+const UNIQUE_TEL_TRUTHY_VALUES = new Set([ 'true', 'yes', 'true()' ]);
```
Comment (dianabarsan) [inline webapp/src/js/enketo/widgets/phone-widget.js]: In this comment: https://github.com/medic/cht-core/issues/10751#issuecomment-4120102325
@jkuester mentioned that these values are generically accepted pyxform yes/no. 

So for completeness, I think we should add this "truthy" evaluator function in a new library: https://github.com/medic/cht-core/tree/1afebb7970d5d92afcd4c709921410fbf741dfaf/webapp/src/js/enketo/lib, because they don't only apply to phone uniqueness fields. 
It would also be helpful to link to the pyxform source code in a comment to persist the connection.

Comment (vikrantwiz02) [reply webapp/src/js/enketo/widgets/phone-widget.js]: Thank you so much for the guidance, @dianabarsan. I’m really grateful for the feedback on the architecture—it’s a great learning experience for me to see how these utilities should be centralized.

I have implemented the changes exactly as you suggested:
- Library Refactor: Moved the truthy/falsy logic into webapp/src/js/enketo/lib/pyxform-boolean.js.
- Documentation: Included the official pyxform source reference in the comments to ensure we stay aligned with the ODK standard.
- History Cleanup: Flattened the branch into a single clean commit on top of the latest master to remove the previous merge noise.

Testing Results:

- I ran the full Enketo regression in webapp/: npm run unit -- --watch=false --include='../tests/karma/js/enketo/**/*.spec.ts'
- Result: 103/103 tests passed (including the new suite for the pyxform-boolean library and the updated phone widget tests).

Thank you again for your time and patience with this PR. Ready for your review!

File: webapp/tests/karma/js/enketo/widgets/phone-widget.spec.ts [hunk 2]
```diff
@@ -24,8 +24,8 @@ describe('Enketo: Phone Widget', () => {
   const proxySelector = (name) => inputSelector(name).prev();
 
 
-  const buildHtml = (chtUniqueTel?) => {
-    const chtUniqueTelData = chtUniqueTel ? `data-cht-unique_tel="${chtUniqueTel}"` : '';
+  const buildHtml = (chtUniqueTel = '__ATTRIBUTE_NOT_SET__') => {
+    const chtUniqueTelData = chtUniqueTel === '__ATTRIBUTE_NOT_SET__' ? '' : `data-cht-unique_tel="${chtUniqueTel}"`;
```
Comment (dianabarsan) [inline webapp/tests/karma/js/enketo/widgets/phone-widget.spec.ts]: Can you please explain the reasoning behind this change? 
I usually don't like using magic strings, especially repeatedly, so I'm wondering if this change is necessary.

**Comment by vikrantwiz02** [comment]
Thank You @dianabarsan 

I originally used that magic string to strictly differentiate between a test explicitly passing an empty string (buildHtml('')) versus not passing the argument at all (buildHtml()), since this PR specifically tests various truthy/falsy values.

However, you are completely right that it's an anti-pattern and I was overcomplicating it. I've updated the test helper to make the parameter optional (chtUniqueTel?: string) and check natively against undefined, which handles the distinction much more cleanly.

In this latest push, I have:
- Removed the magic string in favor of the undefined check.
- Fixed the ESLint formatting error.
All unit tests and lint checks are completely green locally now! Let me know if you need anything else.

**Comment by vikrantwiz02** [comment]
@dianabarsan, I re-ran the failing jobs separately on my fork, and both passed successfully:
[https://github.com/vikrantwiz02/cht-core/actions/runs/23653223782](url)
So if they fail here again, re-running those specific jobs should likely fix it.

Thanks and if any further updates or changes are needed, please let me know, I’d be happy to help.

**Comment by vikrantwiz02** [comment]
Thank you for the guidance and the review, @dianabarsan! I see all the tests have finally passed now. Looks like it is ready to merge whenever you have a moment.
